resfulservice/src/graphql/resolver/material_template/mutation.js

```diff
@@ -30,6 +30,24 @@ const materialMutation = {
     } catch (error) {
       return errorFormater(error.message, 500);
     }
+  },
+
+  deleteXlsxCurationList: async (_, { input }, { user, req, isAuthenticated }) => {
+    req.logger.info('deleteMaterialColumn Function Entry:');
+    if (!isAuthenticated) {
+      req.logger?.error('[deleteMaterialColumn]: User not authenticated to remove listing');
+      return errorFormater('not authenticated', 401);
+    }
+    const { field } = input;
+    try {
+      const columnExists = await MaterialTemplate.findOne({ field });
+      req.logger?.info('columnExists: ');
+      req.logger?.info(columnExists);
+      if (!columnExists) return errorFormater('column not found', 404);
+      return MaterialTemplate.findOneAndDelete({ field }).lean();
```

Comment:
I'll suggest bringing this 
```
req.logger?.info('columnExists: ');
req.logger?.info(columnExists);
```
below this call to get if it matches anything - log and return else log error and return

Reply:
The problem I'm having here is that findOneAndDelete just returns null if nothing is found (doesn't error out), so I might still need to check for null before returning and manually send the 404

Reply:
Sure, it will return null for invalid matches. I would suggest an if statement check.